### ISSUE TITLE
Add a Podcast Episode bug

### ISSUE THREAD

**Issue by glj** [BabDev/Podcast-Manager#178]
Joomla 3.3.1. These issues occur when a registered user tries to submit a podcast in the site front end. Adding podcasts using the Podcast Manager component in the backend works well.

Issue 1. The menu item of type "Add a Podcast Episode" does not display correctly the Created By area in the Publishing Options section as shown below.

![pm-error1](https://cloud.githubusercontent.com/assets/8038867/3638926/23ca1c0a-1069-11e4-9846-79caa96f41ec.jpg)

Also, clicking on the Select User box displays the following instead of a user list.

![pm-error2](https://cloud.githubusercontent.com/assets/8038867/3638934/8823b7e2-1069-11e4-8cd1-b2e35d9684de.jpg)

Issue 2. After supplying the required data in the Podcast Metadata section, clicking on the Save button gives the following error:

Catchable fatal error: Argument 1 passed to PodcastManagerControllerPodcast::postSaveHook() must be an instance of JModel, instance of PodcastManagerModelPodcast given, called in /home/.../public_html/libraries/legacy/controller/form.php on line 829 and defined in /home/../public_html/components/com_podcastmanager/controllers/podcast.php on line 312

Interestingly, in spite of the nasty looking error message, the submitted podcast is indeed submitted.

**Comment by jeremejazz** [comment]
Hi sorry to open this again. I am still experiencing this in Joomla 3.5 on the front-end part. Add episode still works but goes to blank screen. The file is uploaded but the problem is a fatal error is displayed:

`Catchable fatal error: Argument 1 passed to PodcastManagerControllerPodcast::postSaveHook() must implement interface JModel, instance of PodcastManagerModelPodcast given, called in C:\wamp64\www\joomla_...\libraries\legacy\controller\form.php on line 833 and defined in C:\wamp64\www\joomla_...\components\com_podcastmanager\controllers\podcast.php on line 312
`

Is there a way to patch this up? Or maybe I can edit something on this . Joomla 2.5 is no longer supported as of this time right?
Thank you.

**Comment by jeremejazz** [comment]
I have tried checking the error. Not sure this is a good idea but I have removed the class/interface type declaration in the postSaveHook function located in \components\com_podcastmanager\controllers\podcast.php on line 312. And now it redirects after saving. I think there is a better idea for this but for now it works. Don't want fellow authors to get freaked or double post since refreshing submits the data again.  :+1:
